test/unittests/state_validate_transaction_test.cpp

```diff
@@ -0,0 +1,42 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <test/state/errors.hpp>
+#include <test/state/state.hpp>
+
+using namespace evmone::state;
+
+TEST(transaction, validate_transaction)
```

Comment:
```suggestion
TEST(state_tx, validate_nonce)
```
Also change file name to `state_tx_test.cpp`